### ISSUE TITLE
Missed Audio Codecs on Latest Build!

### ISSUE THREAD

**Issue by jicomy** [BtbN/FFmpeg-Builds#138]
hello and i hope you have good time

when i downloaded latest build of [https://www.gyan.dev/ffmpeg/builds/](https://www.gyan.dev/ffmpeg/builds/)
and i checked codecs and i found out that some of audio encoder not exist in your latest build

here a list of removed/missed codec that exist in other people builds but not in BtbN build (tested on [Latest Auto-Build (2022-03-13 12:33)](https://github.com/BtbN/FFmpeg-Builds/releases/tag/latest)
**amr_wb** Encoder
**gsm** Encoder
**gsm_ms** Encoder
**ilbc** Encoder
**speex** Encoder

**Comment by BtbN** [comment]
Are those external libraries?
Everything that has a native de/encoder in ffmpeg should be enabled by default.

**Comment by jicomy** [comment]
yes, all of them using libraries, but thats not the point, because vorbis and mpeg3 using libraries too
this codecs that i reported are exist in all of other peoples builds

**Comment by BtbN** [comment]
Yes, Vorbis and mp3 are also hugely popular, while these ones are ultra rare, so it's a tradeoff between maintenance work needed, increase in binary size, and increase in attack surface.
Do you personally need any of those, or are you just comparing builds?

**Comment by jicomy** [comment]
yes, i need gsm_ms one
when i change my ffmpeg.exe with this BtbN Build, i found out that something is wrong because i had it before now there is no such codec when i changed from gyan.dev build to BtbN build

**Comment by BtbN** [comment]
libgsm looks archaic, I'll see if I can get it working without too much trouble.
It's packaged in basically all distros, so there's a bunch of examples to work with.

**Comment by jicomy** [comment]
its archaic, but some people, someday, at some place or corner will need them, however, if you enable all of missed libraries it will be great, because FFmpeg mainly used to be as an converter, however, it does much more
maybe there is more codecs that their encoders are available in FFmpeg but not included in any build, enabling them will be great
Thank you for Answering, Have Good time

**Comment by BtbN** [comment]
It very much depends on the library. Some of them don't play nice with modern compilers, OS, or static linking.
Or are such a huge risk that it's not worth to enable.
libgsm looks relatively fine though, just has a weird build system that needs to be dealt with somehow.

**Comment by theraininsky** [comment]
**BtbN missing the following encoders:**
Hap
libvo_amrwbenc
libgsm
libgsm_ms
libilbc
libshine
libspeex
**While gyan missing**
libopenh264
libjxl

**Comment by klaxons1** [comment]
any progress on this? libvo_amrwbenc is still mising

**Comment by BtbN** [comment]
Pull requests are always welcome.
Every addition is a compromise between making the binary even bigger and its usefulness, so the more niche something is, the less motivated I am to a) put time into it and b) make the already quite large binaries even bigger.
